suite/tests/api/ir_aarch64.c

```diff
@@ -94,6 +98,169 @@ test_extend(void *dc)
     test_base_disp_extend(DR_EXTEND_SXTW, false, OPSZ_4, 0);
 }
 
+#define CHECK_INSTR(opcode) \
+    ASSERT(instr_get_opcode(instr) == opcode); \
+    instr_disassemble(dc, instr, STDOUT); print("\n"); \
+    pc = instr_encode(dc, instr, buf); \
+    ASSERT(pc != NULL);
```

Comment:
In addition to checking pc != null, could we also decode the encoded instruction and it should match what we put in